internal/common/armadaerrors/errors_test.go

```diff
@@ -0,0 +1,155 @@
+package armadaerrors
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/G-Research/armada/internal/common/requestid"
+)
+
+func TestCodeFromError(t *testing.T) {
+	tests := map[string]struct {
+		err  error
+		want codes.Code
+	}{
+		"ErrAlreadyExists":                {&ErrAlreadyExists{}, codes.AlreadyExists},
+		"ErrNotFound":                     {&ErrNotFound{}, codes.NotFound},
+		"ErrInvalidArgument":              {&ErrInvalidArgument{}, codes.InvalidArgument},
+		"pkg.Error => ErrAlreadyExists":   {errors.WithMessage(&ErrAlreadyExists{}, "foo"), codes.AlreadyExists},
+		"pkg.Error => ErrNotFound":        {errors.WithMessage(&ErrNotFound{}, "foo"), codes.NotFound},
+		"pkg.Error => ErrInvalidArgument": {errors.WithMessage(&ErrInvalidArgument{}, "foo"), codes.InvalidArgument},
+		"pkg.Error":                       {errors.New("foo"), codes.Unknown},
+		"nil":                             {nil, codes.OK},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := CodeFromError(tc.err)
+			if got != tc.want {
+				t.Fatalf("expected %v but got %v", tc.want, got)
```

Comment:
Style 

Most of the project would use:
`assert.Equal(t, tc.want, got, "expected %v but got %v", tc.want, got)`

Any reason we are swapping from that?